src/imagery/i.sentinel/i.sentinel3.import/i.sentinel3.import.py

```diff
@@ -0,0 +1,844 @@
+#%end
+
+#%option
+#% key: modified_before
+#% description: Import only files modified before this date ("YYYY-MM-DD")
+#% guisection: Filter
+#%end
+
+#%option G_OPT_F_OUTPUT
+#% key: register_output
+#% description: Name for output file to use with t.register
+#% required: no
+#%end
+
+#%option G_OPT_M_DIR
+#% key: metadata
+#% description: Name of directory into which Sentinel metadata json dumps are saved
+#% required: no
+#%end
+
+#%option
+#% key: nprocs
+#% description: Number of parallel processes to use during import (default=1)
+#% type: integer
+#% answer: 1
+#% required: no
+#%end
+
+#%option
+#% key: memory
+#% type: integer
+#% required: no
+#% multiple: no
+#% label: Maximum memory to be used (in MB)
+#% description: Cache size for raster rows
+#% answer: 300
+#%end
+
+#%flag
+#% key: a
+#% description: Apply cloud mask before import (can significantly speed up import)
+#% guisection: Settings
+#%end
+
+#%flag
+#% key: c
+#% description: Import LST in degree celsius (default is kelvin)
+#% guisection: Settings
+#%end
+
+#%flag
+#% key: p
+#% description: Print raster data to be imported and exit
+#% guisection: Print
+#%end
+
+#%flag
+#% key: j
+#% description: Write meta data json for each band to LOCATION/MAPSET/cell_misc/BAND/description.json
+#% guisection: Print
+#%end
+
+#%rules
+#% excludes: -p,register_output
+#%end
+
+import atexit
+from datetime import datetime
+from itertools import chain
+import os
+import sys
+from pathlib import Path
+import json
+from zipfile import ZipFile
+from multiprocessing import Pool
+
+import grass.script as gscript
+from grass.pygrass.modules import Module, MultiModule
+from grass.temporal.datetime_math import (
+    datetime_to_grass_datetime_string as grass_timestamp,
+)
+
+
+TMPFILE = None
+
+
+def cleanup():
+    # remove temporary maps
+    if TMPFILE is not None:
+        gscript.try_remove(TMPFILE)
+
+
+def np_as_scalar(var):
+    if type(var).__module__ == np.__name__:
+        if var.size > 1:
+            return str(var)
+        return var.item()
+    else:
+        return var
+
+
+def filter(input_dir, pattern, modified_after, modified_before):
+    # Filter files according to pattern
+    if not pattern:
+        pattern = "S3*.zip"
+
+    s3_files = []
+    s3_files = list(input_dir.glob(pattern))
+
+    if len(s3_files) > 0 and (
+        modified_after is not None or modified_before is not None
+    ):
+        modified_after = modified_after if modified_after is not None else datetime.min
+        modified_before = (
+            modified_before if modified_before is not None else datetime.now()
+        )
+        s3_files = [
+            s3_file
+            for s3_file in s3_files
+            if modified_after
+            < datetime.fromtimestamp(s3_file.stat().st_mtime)
+            < modified_before
+        ]
+
+    if len(s3_files) < 1:
+        gscript.fatal(
+            _("Nothing found to import. Please check input and pattern_file options.")
```

Comment:
pattern_file seems a leftover of i.sentinel.import, right?

Reply:
Yes, thanks for pointing that out.
Since bands are not filtered by a regex here, I used only `pattern` as option name for filtering files. But I now also see that checks of the pattern and modification dates should be done differently...